Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -306,35 +308,23 @@ void ProcessThrottler::updateThrottleStateIfNeeded()
     updateThrottleStateNow();
 }
 
-#if USE(EXTENSIONKIT_ASSERTIONS)
-void ProcessThrottler::didConnectToProcess(RetainPtr<_SEExtensionProcess> process)
+void ProcessThrottler::didConnectToProcess(AuxiliaryProcessProxy& process)
 {
     PROCESSTHROTTLER_RELEASE_LOG("didConnectToProcess");
     RELEASE_ASSERT(!m_assertion);
 
-    m_process = process;
+    m_processProxy = WeakPtr { process };
     updateThrottleStateNow();
     RELEASE_ASSERT(m_assertion || (m_state == ProcessThrottleState::Suspended && !m_shouldTakeNearSuspendedAssertion));
 }
-#else
-void ProcessThrottler::didConnectToProcess(ProcessID pid)
-{
-    PROCESSTHROTTLER_RELEASE_LOG_WITH_PID("didConnectToProcess:", pid);
-    RELEASE_ASSERT(!m_assertion);
-
-    m_processID = pid;
-    updateThrottleStateNow();
-    RELEASE_ASSERT(m_assertion || (m_state == ProcessThrottleState::Suspended && !m_shouldTakeNearSuspendedAssertion));
-}
-#endif
 
 void ProcessThrottler::didDisconnectFromProcess()
 {
-    PROCESSTHROTTLER_RELEASE_LOG_WITH_PID("didDisconnectFromProcess:", m_processID);
+    PROCESSTHROTTLER_RELEASE_LOG("didDisconnectFromProcess:");
```

Comment:
I hate the idea of losing this logging. Is there a way to determine the PID from the m_processProxy so we can keep this useful diagnostic information?

Reply:
Maybe:
`PROCESSTHROTTLER_RELEASE_LOG_WITH_PID("didDisconnectFromProcess:", m_processProxy  ? m_processProxy->processID() : 0);`

Reply:
The macro `PROCESSTHROTTLER_RELEASE_LOG` actually logs the PID, so I believe we are good here.